### DISCUSSION TITLE
The current huskedspeciesname limitations feel very unnecessary

### DISCUSSION THREAD

**Discussion by washedAwry** [FakeFishGames/Barotrauma#14299] [Code questions and modding support]
For a while, huskedspeciesname would allow you to either turn all affected targets into one species (i.e if you wanted a zombie infection to somehow take over a humanhusk as well as humans while resulting in the same kind of zombie) or into a spectrum of species (i.e [speciesname]husk, having humans turning into humanhusk and crawlers turning into crawlerhusk)

I'm not certain of the update in which it changed, but several months ago it began to require a specific pattern rather than the modder being able to choose whether or not they use [speciesname] (for, say, a broad infection that makes infected creatures of various forms like the husk infection). This makes alieninfections a lot less flexible than they used to, and outright prevents certain once-possible interactions, like having two infections fight over hosts in a sickly game of tug-of-war or having a spreading fungal infection that affects and grows over nearly any humanoid creature, modded or unmodded.
**"Note that all the infected speciesnames and files must stick the following pattern: [nonhuskedspeciesname][huskedspeciesname]. E.g. Humanhusk, Crawlerhusk, or Humancustomhusk, or Crawlerzombie. Not "Customhumanhusk!" or "Zombiecrawler""**
This single, small change bricked so many of my mods that it completely killed my motivation to work on anything for months because of how reliant on this behavior many of them were. 

If someone could please offer an explanation as to why this change occurred in the first place, and if it is possible add an optional attribute to afflictions to revert huskedspeciesname to the old logic, that would be greatly appreciated. 
I am willing to adapt my methods if similar behavior can be replicated, but would prefer being able to apply the prior logic to my mods.

**Comment by ubertpendragon** [comment]
You also cannot make variants of say CrawlerHusk, the game won't like it
